File: getNodesUplinkData.py
from http.server import HTTPServer, BaseHTTPRequestHandler
from urllib.parse import urlparse, parse_qs
import os
from chirpstack_api import integration
from google.protobuf.json_format import Parse

os.environ['PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION'] = 'python'

class Handler(BaseHTTPRequestHandler):
    def do_POST(self):
        self.send_response(200)
        self.end_headers()
        query_args = parse_qs(urlparse(self.path).query)

        content_len = int(self.headers.get('Content-Length', 0))
        body = self.rfile.read(content_len)

        if query_args["event"][0] == "up":
            self.up(body)
        else:
            print("handler for event %s is not implemented" % query_args["event"][0])

    def up(self, body):
        up = self.unmarshal(body, integration.UplinkEvent())
        print("Uplink received with payload: %s" % (up))

    def unmarshal(self, body, pl):
        return Parse(body, pl, ignore_unknown_fields=True)

httpd = HTTPServer(('', 8090), Handler)
httpd.serve_forever()

File: getNodesUplinkData.js
const http = require('http');
const url = require('url');

const server = http.createServer((req, res) => {
    if (req.method === 'POST') {
        let body = '';
        req.on('data', chunk => {
            body += chunk.toString();
        });
        req.on('end', () => {
            res.writeHead(200);
            res.end();

            const query = url.parse(req.url, true).query;

            if (query.event === 'up') {
                console.log(`Uplink received with payload: ${body}`);
            } else {
                console.log(`handler for event ${query.event} is not implemented`);
            }
        });
    }
});

server.listen(8090);
